libraries/botbuilder-applicationinsights/tests/telemetryClient.test.js

```diff
@@ -0,0 +1,238 @@
+
+            client.trackTrace(mockArgument);
+
+            assert(spy.calledOnce);
+        });
+
+        it('trackPageView() method should call the same method in default appInsights client', function() {
+            const client = new ApplicationInsightsTelemetryClient('fakeKey');
+            const spy = sinon.spy(appInsights.defaultClient, 'trackPageView');
+            const mockArgument = {};
+
+            client.trackPageView(mockArgument);
+
+            assert(spy.calledOnce);
+        });
+
+        it('flush() method should call the same method in default appInsights client', function() {
+            const client = new ApplicationInsightsTelemetryClient('fakeKey');
+            const spy = sinon.spy(appInsights.defaultClient, 'flush');
+
+            client.flush();
+
+            assert(spy.calledOnce);
+        });
+
+        it('addBotIdentifiers() correctly sets App Insights values', function() {
+            const client = new ApplicationInsightsTelemetryClient('fakeKey');
+            const activity = {
+                from: { id: 'stub from id' },
+                channelId: 'stub channel id',
+                conversation: { id: 'stub conversation id' },
+                id: 'stub activity id',
+                type: 'stub activity type'
+            };
+            const operation = { 
+                id: 'stub operation id',
+                name: 'stub operation name',
+                parentId: 'stub parent id'
+            };
+            const stub = sinon.stub(CorrelationContextManager, 'getCurrentContext');
+            stub.returns({
+                activity,
+                operation 
+            });
+            const spy = sinon.spy(client.client.channel, 'send');
+
+            client.trackEvent({});
+
+            const envelope = spy.args[0][0];
+            assert.strictEqual(envelope.iKey, 'fakeKey');
+            assert.strictEqual(envelope.tags['ai.operation.id'], operation.id);
+            assert.strictEqual(envelope.tags['ai.operation.parentId'], operation.parentId);
+            assert.strictEqual(envelope.tags['ai.user.id'], `${ activity.channelId }${ activity.from.id }`);
+            assert.strictEqual(envelope.data.baseData.properties.activityId, activity.id);
+            assert.strictEqual(envelope.data.baseData.properties.activityType, activity.type);
+            assert.strictEqual(envelope.data.baseData.properties.channelId, activity.channelId);
+            assert.strictEqual(envelope.data.baseData.properties.conversationId, activity.conversation.id);
+        });
+
+        it('addBotIdentifiers() correctly sets App Insights values, defaulting to empty strings', function() {
+            const client = new ApplicationInsightsTelemetryClient('fakeKey');
+            client.client.commonProperties = undefined;
+            const activity = {
+                from: undefined,
+                channelId: undefined,
+                conversation: undefined,
```

Comment:
Shouldn't these be unnecessary? Or is something doing an Object.keys()/hasOwnProperty?

Reply:
It's unnecessary. Since that test tests that those properties become empty string, I was just stating explicitly that the activity starts with those undefined. But yeah, functionally it's unnecessary, so I'll remove it.